### ISSUE TITLE
Perl debugger non-functional in core-p7 branch

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#14]
The debugger is gagging for prototype/signature-related reasons. Consider:
```
$ ./perl -Ilib t/base/cond.t
1..4
ok 1 - operator eq
ok 2 - operator ne
ok 3 - operator ==
ok 4 - operator !=

 $ ./perl -d -Ilib t/base/cond.t
A signature parameter must start with '$', '@' or '%' at lib/perl7db.pl line 875, near "(*"
syntax error at lib/perl7db.pl line 875, near "(*"
A signature parameter must start with '$', '@' or '%' at lib/perl7db.pl line 876, near "(\"
syntax error at lib/perl7db.pl line 876, near "(\["
BEGIN not safe after errors--compilation aborted at lib/perl7db.pl line 876.
Compilation failed in require.
BEGIN failed--compilation aborted.
```
This impedes development.  What can be done?

Thank you very much.
Jim Keenan

**Comment by atoomic** [comment]
this is fixed by 394bf9330072fcbaf38a988a8e5ac620f7005dd6

**Comment by jkeenan** [comment]
Ummmm, not quite.  It didn't immediately crash, but ...
```
$ gitcurr
core-p7

$ git describe
v5.32.0-564-g4efb837c23

$ ./perl -d -Ilib t/base/cond.t

Loading DB routines from perl5db.pl version 1.57
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.

main::(t/base/cond.t:5):	print "1..4\n";
Bareword found where operator expected at lib/Term/ReadLine.pm line 379, near "Tgetent Term::Cap"
 at lib/Term/ReadLine.pm line 379.
	require Term/ReadLine.pm called at lib/perl7db.pl line 6827
	DB::setterm() called at lib/perl7db.pl line 1849
	DB::_DB__read_next_cmd(undef) called at lib/perl7db.pl line 2789
	DB::DB called at t/base/cond.t line 5
	(Do you need to predeclare Tgetent?)
 at lib/Term/ReadLine.pm line 379.
	require Term/ReadLine.pm called at lib/perl7db.pl line 6827
	DB::setterm() called at lib/perl7db.pl line 1849
	DB::_DB__read_next_cmd(undef) called at lib/perl7db.pl line 2789
	DB::DB called at t/base/cond.t line 5
syntax error at lib/Term/ReadLine.pm line 379, near "Tgetent Term::Cap "
syntax error at lib/Term/ReadLine.pm line 397, near "}"
Bareword "LoadTermCap" not allowed while "strict subs" in use at lib/Term/ReadLine.pm line 389.
Compilation failed in require at lib/perl7db.pl line 6827.
 at lib/perl7db.pl line 6827.
	DB::setterm() called at lib/perl7db.pl line 1849
	DB::_DB__read_next_cmd(undef) called at lib/perl7db.pl line 2789
	DB::DB called at t/base/cond.t line 5
Attempt to reload Term/ReadLine.pm aborted.
Compilation failed in require at lib/perl7db.pl line 6827.
END failed--call queue aborted.
 at t/base/cond.t line 0.
```

**Comment by jkeenan** [comment]
Ticket needs to be re-opened.

**Comment by atoomic** [comment]
the error displayed here is not a perl7db issue IMO,  but a bug in lib/Term/ReadLine.pm which is using indirect
going to fix it

**Comment by atoomic** [comment]
that error from `lib/Term/ReadLine.pm`  is fixed by 2785301a
please confirm and close

```
╰─> ./perl -d -Ilib t/base/cond.t

Loading DB routines from perl5db.pl version 1.57
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.

main::(t/base/cond.t:5):	print "1..4\n";
```

**Comment by jkeenan** [comment]
> that error from `lib/Term/ReadLine.pm` is fixed by [2785301](https://github.com/atoomic/perl/commit/2785301ad167ec22e0c0eeb5bf9b13c86022a434)
> please confirm and close
> 

Looks good.

```
$ uname -mrs
FreeBSD 11.2-STABLE amd64

$ gitcurr
core-p7

$ git describe
v5.32.0-566-g2785301ad1

$ ./perl -Ilib -V:config_args
config_args='-des -Dusedevel';

$ ./perl -d -Ilib t/base/cond.t

Loading DB routines from perl5db.pl version 1.57
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.

main::(t/base/cond.t:5):	print "1..4\n";
  DB<1> c
1..4
ok 1 - operator eq
ok 2 - operator ne
ok 3 - operator ==
ok 4 - operator !=
Debugged program terminated.  Use q to quit or R to restart,
use o inhibit_exit to avoid stopping after program termination,
h q, h R or h o to get additional info.
  DB<1> q
```
Thanks for your quick response.
